Bug fixes

iamshital opened the pull request:
1. Fixed issue, where windows was failing to access the files if file path is more than 255 characters.
2. Added "ppa" kernel installation support to CustomKernelInstall.sh [Packages in “PPA for Canonical Kernel Team”](https://launchpad.net/~canonical-kernel-team/+archive/ubuntu/ppa/+packages)
3. Fixed issue of duplicate test name in test summary
4. Split VERIFY-DEPLOYMENT-PROVISION stage in four subtests -
Example: 
VERIFY-DEPLOYMENT-PROVISION Execution Time: 9.92 minutes
	VERIFY-DEPLOYMENT-PROVISION : PASS  
		  FirstBoot : PASS 
		  FirstBoot : Call Trace Verification : PASS 
		  Reboot : PASS 
		  Reboot : Call Trace Verification : PASS 
5. Fixed Telemetry column name
6. Added VERIFY-DEPLOYMENT-PROVISION-SRIOV test

ader1990 commented on Testscripts/Linux/customKernelInstall.sh:
```diff
@@ -122,6 +122,54 @@ InstallKernel()
                         UpdateTestState $ICA_TESTCOMPLETED
                 fi
+        elif [ "${CustomKernel}" == "proposed" ]; then
+                DISTRO=`grep -ihs "buntu\|Suse\|Fedora\|Debian\|CentOS\|Red Hat Enterprise Linux" /etc/{issue,*release,*version}`
```
put this in a method and reuse it (duplicate code)

ader1990 commented on Testscripts/Linux/customKernelInstall.sh:
```diff
@@ -122,6 +122,54 @@ InstallKernel()
+                then
+                        LogMsg "Enabling proposed repositry..."
+                        echo "deb http://archive.ubuntu.com/ubuntu/ xenial-proposed restricted main multiverse universe" >> /etc/apt/sources.list
+                        rm -rf /etc/apt/preferences.d/proposed-updates
```
always use quotes when working with strings (this applies for all current and future code)

ader1990 commented on Testscripts/Linux/customKernelInstall.sh:
```diff
@@ -122,6 +122,54 @@ InstallKernel()
+                fi
+                UpdateTestState $ICA_TESTCOMPLETED
+                if [ $kernelInstallStatus -ne 0 ]; then
+                        LogMsg "CUSTOM_KERNEL_FAIL"
```
keep the indentation consistent (never use tabs or more or less than 4 spaces)

ader1990 commented on RunTests.ps1:
```diff
@@ -67,7 +67,26 @@ Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1"
-    $WorkingDirectory = Split-Path -parent $MyInvocation.MyCommand.Definition 
+    $MaxDirLength = 32
+    $WorkingDirectory = Split-Path -parent $MyInvocation.MyCommand.Definition
+    Set-Variable -Name shortRandomNumber -Value $(Get-Random -Maximum 99999 -Minimum 11111) -Scope Global
```
create a function for the random number logic

ader1990 commented on RunTests.ps1:
```diff
@@ -67,7 +67,26 @@ Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1"
+    $MaxDirLength = 32
+    $WorkingDirectory = Split-Path -parent $MyInvocation.MyCommand.Definition
+    Set-Variable -Name shortRandomNumber -Value $(Get-Random -Maximum 99999 -Minimum 11111) -Scope Global
+    Set-Variable -Name shortRandomWord -Value $(-join ((65..90) | Get-Random -Count 4 | ForEach-Object {[char]$_})) -Scope Global
```
create a function for the random word logic

ader1990 commented on RunTests.ps1:
```diff
@@ -67,7 +67,26 @@ Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1"
+        Write-Host "Copying current workspace to $finalWorkingDirectory"
+        Copy-Item -Path $tmpSource -Destination $finalWorkingDirectory -Recurse -Force -ErrorAction SilentlyContinue| Out-Null
+        Set-Location -Path $finalWorkingDirectory | Out-Null
+        Write-Host "Wroking directory changed to $finalWorkingDirectory"
```
typo

ader1990 commented on RunTests.ps1:
```diff
@@ -67,7 +67,26 @@ Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1"
+        $TempWorkspace = "$(Split-Path $OriginalWorkingDirectory -Qualifier)"
+        New-Item -ItemType Directory -Path "$TempWorkspace\LISAv2" -Force -ErrorAction SilentlyContinue | Out-Null
+        New-Item -ItemType Directory -Path "$TempWorkspace\LISAv2\$shortRandomWord$shortRandomNumber" -Force -ErrorAction SilentlyContinue | Out-Null
+        $finalWorkingDirectory = "$TempWorkspace\LISAv2\$shortRandomWord$shortRandomNumber"
```
never ever hardcode the repo name inside the repo code, it is a huge problem when you want to fork a repo as the repo name might change at fork time and you end up with an unusable code

juhlee-microsoft replied:
I see file copying and folder removal in 'finally' section. This should not remain unused code. I think this is ok.

ader1990 commented on RunTests.ps1:
```diff
@@ -67,7 +67,26 @@ Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1"
+        $finalWorkingDirectory = "$TempWorkspace\LISAv2\$shortRandomWord$shortRandomNumber"
+        $tmpSource = '\\?\' + "$OriginalWorkingDirectory\*"
+        Write-Host "Copying current workspace to $finalWorkingDirectory"
+        Copy-Item -Path $tmpSource -Destination $finalWorkingDirectory -Recurse -Force -ErrorAction SilentlyContinue| Out-Null
```
this is a vital operation and the erroraction should not be set to silently continue

juhlee-microsoft commented:
I am in the same page with @ader1990. Please fix the code along with his comment and let me know. Will sign off then.

iamshital commented:
@juhlee-microsoft I've addressed some issues mentioned by Adrian and created a new issue https://github.com/LIS/LISAv2/issues/93 and assigned it to myself. 
Remaining changes needs code evaluation and proper testing.

juhlee-microsoft commented on Libraries/Framework.psm1:
```diff
@@ -917,8 +917,8 @@ Function UploadTestResultToDatabase ($TestPlatform,$TestLocation,$TestCategory,$
 				$dataTableName = "LISAv2Results"
-				$SQLQuery = "INSERT INTO $dataTableName (DateTimeUTC,TestPlatform,TestLocation,TestCategory,TestArea,TestName,TestResult,SubTestName,SubTestResult,TestTag,GuestDistro,KernelVersion,LISVersion,HostVersion,VMSize,Networking,ARMImage,OsVHD,LogFile,BuildURL) VALUES "
-				$SQLQuery += "('$DateTimeUTC','$TestPlatform','$TestLocation','$TestCategory','$TestArea','$TestName','$testResult','','','$TestTag','$GuestDistro','$KernelVersion','$LISVersion','$HostVersion','$VMSize','$Networking','$ARMImageName','$OsVHD','$UploadedURL', '$BuildURL'),"
+				$SQLQuery = "INSERT INTO $dataTableName (DateTimeUTC,TestPlatform,TestLocation,TestCategory,TestArea,TestName,TestResult,SubTestName,SubTestResult,ExecutionTag,GuestDistro,KernelVersion,LISVersion,HostVersion,VMSize,Networking,ARMImage,OsVHD,LogFile,BuildURL) VALUES "
```
Only verified Field name fixed. It is ok for now. Others go to #93

juhlee-microsoft commented:
Decided to merge to master branch in 7/19 Open Source Automation and Release team Meeting
